boutdata/restart.py

```diff
@@ -609,6 +626,47 @@ def redistribute(npes, path="data", nxpe=None, output=".", informat=None, outfor
         for i in range(npes):
             ix = i % nxpe
             iy = int(i/nxpe)
+
+            def get_block(data):
```

Comment:
A function definition inside the `for` loop looks dangerous.
Will there not be any problems with name shadowing of `ix` and `iy`?

Reply:
If I understand correctly, Python functions are allowed to access variables from the local scope where they are defined, as long as there is not a variable defined in the function scope that shadows the name (see example below). I find the code easier to read when defining the function in the most local scope possible, rather than defining elsewhere and passing a bunch of local variables in as arguments. It doesn't look dangerous to me, but maybe I'm missing something?

example:
```
In [1]: a = 1

In [2]: def foo():
   ...:     print(a)
   ...: 

In [3]: foo()
1

In [4]: def foo2():
   ...:     a = 2
   ...:     print(a)
   ...: 

In [5]: foo2()
2

In [6]: print(a)
1

In [7]: def foo3():
   ...:     print(a)
   ...:     a = 3
   ...:     print(a)
   ...: 

In [8]: foo3()
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-8-1355e5d69dc1> in <module>
----> 1 foo3()

<ipython-input-7-1b05d455e79f> in foo3()
      1 def foo3():
----> 2     print(a)
      3     a = 3
      4     print(a)
      5 

UnboundLocalError: local variable 'a' referenced before assignment

```

Reply:
Thanks for the illuminating example 🙂.
I see that also the [google guide](https://google.github.io/styleguide/pyguide.html#s2.6-nested) accept nestedness, and points out it may be problematic with pickling and testing